django/contrib/admin/views/main.py

```diff
@@ -103,6 +112,9 @@ def __init__(
             self.page_num = 1
         self.show_all = ALL_VAR in request.GET
         self.is_popup = IS_POPUP_VAR in request.GET
+        self.add_facets = model_admin.show_facets == ShowFacets.ALWAYS or (
+            model_admin.show_facets == ShowFacets.ALLOW and IS_FACETS_VAR in request.GET
```

Comment:
Enum comparisons should use `is`: https://docs.python.org/3/howto/enum.html#comparisons